docs/adr/0033-content-server-upgrade-to-react.md

```diff
@@ -0,0 +1,128 @@
+
+#### Options
+
+- **Option A: Multiple experiments, no `/beta` route**: Each block of routes will be guarded by separate experiments. If a user is in the experiment, the router will send users to the React version of the page via a paramater.
+- **Option B: Multiple experiments, `/beta` route**: Each block of routes will be guarded by separate experiments. If a user is in the experiment, the router will send users to the React version of the page via `/beta/<original_url>` with a parameter. You can see the proposed work for this [here](https://github.com/mozilla/fxa/pull/14436)
+- **Option C: A single experiment with config settings, and a `/beta` route**: All routes will be guarded by one experiment and separate feature flags per block of routes. In the router file, we will check experiment status as well as feature flag status before send users to the React version of the page via `/beta/<original_url>` with a parameter.
+- **Option D: A single experiment with config settings, and no `/beta` route**: All routes will be guarded by one experiment and separate feature flags per block of routes. In the router file, we will check experiment status as well as feature flag status before send users to the React version of the page via a parameter.
+
+#### Other Considerations
+
+`fxa-content-server` does a lot of things on app startup that Settings does not do, like metrics spin up, recognizing relying parties, OAuth and Sync checks, etc. Because we intend on eventually removing `fxa-content-server`, we need Settings to eventually do these things on app startup instead. When this is implemented, we won't need to access query params provided by content-server.
+
+### GQL Usage
+
+#### When to Use
+
+We want to extend the GQL functionality that already exists in `fxa-settings` due to the advantages GQL offers (see decision outcomes) and currently, FxA has only implemented GQL on authenticated pages.
+
+- **Option A: Use GQL only on authenticated routes**: Use GQL on authenticated routes, and continue using REST for API calls when users are unauthenticated
+- **Option B: Use GQL everywhere we can**: Add an option per mutation or query to talk to the `graphql-api` without being authenticated, allowing us to use it everywhere **except for likely oauth routes**, which could be another can of worms.
+
+#### Other Implications
+
+If we want to use GQL everywhere we can instead of hitting auth-server directly with REST API calls, and likely even if we're more selective, we have to ensure that the `graphql-api` can handle:
+
+1. Emitting metrics
+2. Sending emails
+3. Sending push notifications (SQS for event-broker)
+
+- **Option I: Create an API in auth-server that we hit from `graphql-api`**
```

Comment:
Is this suggesting that we create a new set of endpoints in the auth-server exclusively for graphql-api?

Reply:
It is, I can make that more clear. I'm currently unsure of level of effort for porting these things into a shared location and I'm a bit worried about scope here. It sounds like we want to do this eventually though and it's something SubPlat also wants to do, so I also wonder if it's possible to do any dividing and conquering.